checks/process_test.go

```diff
@@ -25,6 +28,120 @@ func makeProcess(pid int32, cmdline string) *process.FilledProcess {
 	}
 }
 
+func TestProcessMessages(t *testing.T) {
+	p := []*process.FilledProcess{
+		makeProcess(1, "git clone google.com"),
+		makeProcess(2, "mine-bitcoins -all -x"),
+		makeProcess(3, "foo --version"),
+		makeProcess(4, "foo -bar -bim"),
+		makeProcess(5, "datadog-process-agent -ddconfig datadog.conf"),
+	}
+	c := []*containers.Container{
+		makeContainer("foo"),
+		makeContainer("bar"),
+	}
+	// first container runs pid1 and pid2
+	c[0].Pids = []int32{1, 2}
+	c[1].Pids = []int32{3}
+	lastRun := time.Now().Add(-5 * time.Second)
+	syst1, syst2 := cpu.TimesStat{}, cpu.TimesStat{}
+	cfg := config.NewDefaultAgentConfig()
+	sysInfo := &model.SystemInfo{}
+	lastCtrRates := util.ExtractContainerRateMetric(c)
+
+	for i, tc := range []struct {
+		cur, last       map[int32]*process.FilledProcess
+		containers      []*containers.Container
+		maxSize         int
+		blacklist       []string
+		expectedChunks  int
+		totalProcs      int
+		totalContainers int
+	}{
+		// this tests no container case
```

Comment:
nit: I've found it helpful to add comments like this to the test data and use t.Run to create named test cases

Reply:
Cool.....any reference of what you mentioned here?

Reply:
pointed out some code offline